### ISSUE TITLE
TEMPO: When there are numerous projects causing an overwhelming number of spans, the dropdown list in the Span Name becomes very slow

### ISSUE THREAD

**Issue by halleystar** [grafana/grafana#78780]
### What happened?

When there are numerous projects causing an overwhelming number of spans, the dropdown list in the Span Name becomes very slow
![image](https://github.com/grafana/grafana/assets/5334156/c15829b9-c60b-4a87-a1b0-f8ec61546d6b)

### What did you expect to happen?

The span name should be associated with the Resource Service Name. Only the span name in the Resource Service Name is displayed.

### Did this work before?

The span name should be associated with the Resource Service Name. Only the span name in the Resource Service Name is displayed.

### How do we reproduce it?

When there are numerous projects causing an overwhelming number of spans, the dropdown list in the Span Name becomes very slow

### Is the bug inside a dashboard panel?

no

### Environment (with versions)?

Grafana: 10.2.1
OS: linux 
Browser: crhome

### Grafana platform?

None

**Comment by tonypowa** [comment]
hi @halleystar 

Could you record your screen while reproducing the issue?  This will give us an idea of how long it takes for the dropdown to load or become responsive

Thank you

**Comment by aocenas** [comment]
Yeah we don't have any limit here at the moment. We should figure out first if the problem is loading all the options or on the FE showing all the options. If on the FE we can limit what we show but then we have to figure out how to make those option still available. Ideally if the dropdown knows to filter the options based on what user types and that way we can both limit number of options shown and still make them available.

**Comment by aocenas** [comment]
I think we should fix this in some generic way so this does not happen in other options/drowpdowns/datasources cause this seems to be a common issue we have.
